File: Implementation/ClientApplication/src/clientapplication/TravelDetails.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package clientapplication;

import java.util.HashMap;
import java.util.Map;

/**
 *
 * @author hassa
 */
public class TravelDetails {
    String userID;
    String messageID;
    Map<String, String> locationData;
    String date;
    Map<String, String> weatherData;
    
    public TravelDetails(String userID, String messageID, String date, String location, String lat, String lon, String temp, String wind, String rain, String snow, String desc) {
        this.userID = userID;
        this.messageID = messageID;
        this.date = date;
        
        this.locationData = new HashMap<>();
        this.locationData.put("location", location);
        this.locationData.put("latitude", lat);
        this.locationData.put("longitude", lon);
        
        this.weatherData = new HashMap<>();
        this.weatherData.put("tempC", temp);
        this.weatherData.put("windspeedKmh", wind);
        this.weatherData.put("chanceOfRain", rain);
        this.weatherData.put("chanceOfSnow", snow);
        this.weatherData.put("description", desc);
    }
}

File: Implementation/ClientApplicationC/TravelDetails.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClientApplicationC
{
    public class TravelDetails
    {
        public string userID;
        public string messageID;
        public Dictionary<string, string> locationData;
        public string date;
        public Dictionary<string, string> weatherData;

        public TravelDetails(string userID, string messageID, string date, string location, string lat, string lon, string temp, string wind, string rain, string snow, string desc)
        {
            this.userID = userID;
            this.messageID = messageID;
            this.date = date;

            this.locationData = new Dictionary<string, string>();
            this.locationData.Add("location", location);
            this.locationData.Add("latitude", lat);
            this.locationData.Add("longitude", lon);

            this.weatherData = new Dictionary<string, string>();
            this.weatherData.Add("tempC", temp);
            this.weatherData.Add("windspeedKmh", wind);
            this.weatherData.Add("chanceOfRain", rain);
            this.weatherData.Add("chanceOfSnow", snow);
            this.weatherData.Add("description", desc);
        }
    }
}
